cmake/compilerDefinitions.cmake

```diff
@@ -1,10 +1,20 @@
-if (UNIX)
+if (MSVC)
+    # Visual Studio only sets _DEBUG
     if (CMAKE_BUILD_TYPE MATCHES "Debug")
+        add_definitions(-DDEBUG)
+    endif()
+endif()
+if (UNIX)
+    # TODO: check if this can be enabled again for Clang - also done in Makefile
+    if (CMAKE_BUILD_TYPE MATCHES "Debug" AND NOT (CMAKE_CXX_COMPILER_ID MATCHES "Clang"))
         add_definitions(-D_GLIBCXX_DEBUG)
     endif()
     if (HAVE_RULES)
         add_definitions(-DHAVE_RULES -DTIXML_USE_STL)
     endif()
+    if (USE_Z3)
+        add_definitions(-DUSE_Z3 -DNEW_Z3)
```

Comment:
The "NEW_Z3" should be optional somehow.

Reply:
Needs that ticket fixed which actually checks the versions of the z3 library instead of having this define. Since it is currently OFF by default I thought this might be fine for now.

Reply:
Will remove this if the master build is okay again.

Reply:
I removed the NEW_Z3

Reply:
The master doesn't build because it cannot find the `z3_version.h` include - something is still missing.